### ISSUE TITLE
ValueError: time data '24:28:49' does not match format '%H:%M:%S'

### ISSUE THREAD

**Issue by lningyawen** [DataBiosphere/toil#3702]
Hello

I was using cactus-prepare-toil to calculate my data recently. This is the script I ran: 

```
export PATH=/HOME/soft/python3/bin:$PATH
export PATH=/HOME/biosoft/cactus/bin:$PATH
export PYTHONPATH=/HOME/biosoft/cactus/submodules:$PYTHONPATH
source /HOME/biosoft/cactus_env/bin/activate
export TMPDIR=/HOME/cactus_alignment/tmpdir

cd /HOME/cactus_alignment/
mkdir workdir outdir tmpdir

export TOIL_TORQUE_ARGS="-q mem3T"

cactus-prepare-toil  ./jobstore my.seqFile \
  --binariesMode local \
  --batchSystem torque \
  --realTimeLogging \
  --outDir /HOME/cactus_alignment/outdir \
  --workDir /HOME/cactus_alignment/workdir \
  --outHal /HOME/cactus_alignment/outdir/my.hal \
  --disableAutoDeployment \
  --disableCaching \
  --maxNodes 16 \
  --minNodes 1 \
  --preprocessCores 50 \
  --blastCores 50 \
  --alignCores 50 \
  --preprocessMemory 200G \
  --blastMemory 200G \
  --alignMemory 200G \
  --preprocessDisk 2000G \
  --blastDisk 2000G \
  --alignDisk 200G \
  --halAppendDisk 2000G \
  --preprocessPreemptible 50 \
  --blastPreemptible 50 \
  --alignPreemptible 50 \
  --halAppendPreemptible 50 \
  --stats
```

Then save it as a `cactus.sh` file and use `bash cactus.sh` to commit it. However, I encountered the following error: 

```
Version: 6.0.2
Commit: d9a34839a0f975d5c487bbfcf5dcb10b6a8f1e79
[2021-07-06T22:26:47+0800] [MainThread] [W] [toil.common] Batch system does not support auto-deployment. The user script ModuleDescriptor(dirPath='/gpfs/home/liunyw/biosoft/cactus_env/lib/python3.6/site-packages', name='cactus.shared.common', fromVirtualEnv=True) will have to be present at the same location on every worker.
[2021-07-06T22:26:47+0800] [MainThread] [I] [toil.job] Saving graph of 20 jobs, 20 new
[2021-07-06T22:26:47+0800] [MainThread] [I] [toil.job] Processing job 'JobFunctionWrappingJob' kind-JobFunctionWrappingJob/instance-v25lqly0
............................................................................................................................................................................................................
............................................................................................................................................................................................................
collapsed
............................................................................................................................................................................................................
............................................................................................................................................................................................................
[2021-07-07T14:27:12+0800] [MainThread] [I] [toil.leader] 0 jobs are running, 5 jobs are issued and waiting to run
[2021-07-07T15:27:13+0800] [MainThread] [I] [toil.leader] 0 jobs are running, 5 jobs are issued and waiting to run
[2021-07-07T16:27:14+0800] [MainThread] [I] [toil.leader] 0 jobs are running, 5 jobs are issued and waiting to run
[2021-07-07T17:27:14+0800] [MainThread] [I] [toil.leader] 0 jobs are running, 5 jobs are issued and waiting to run
[2021-07-07T18:27:15+0800] [MainThread] [I] [toil.leader] 0 jobs are running, 5 jobs are issued and waiting to run
[2021-07-07T19:27:15+0800] [MainThread] [I] [toil.leader] 0 jobs are running, 5 jobs are issued and waiting to run
[2021-07-07T20:27:16+0800] [MainThread] [I] [toil.leader] 1 jobs are running, 3 jobs are issued and waiting to run
[2021-07-07T20:34:28+0800] [MainThread] [I] [toil.realtimeLogger] Stopping real-time logging server.
[2021-07-07T20:34:28+0800] [MainThread] [I] [toil.realtimeLogger] Joining real-time logging server thread.
Traceback (most recent call last):
  File "/HOME/biosoft/cactus_env/bin/cactus-prepare-toil", line 8, in <module>
    sys.exit(main_toil())
  File "/HOME/biosoft/cactus_env/lib/python3.6/site-packages/cactus/progressive/cactus_prepare.py", line 46, in main_toil
    return main(toil_mode=True)
  File "/HOME/biosoft/cactus_env/lib/python3.6/site-packages/cactus/progressive/cactus_prepare.py", line 230, in main
    cactusPrepare(options, project)
  File "/HOME/biosoft/cactus_env/lib/python3.6/site-packages/cactus/progressive/cactus_prepare.py", line 395, in cactusPrepare
    get_plan(options, project, seqFile, outSeqFile, toil=toil)
  File "/HOME/biosoft/cactus_env/lib/python3.6/site-packages/cactus/progressive/cactus_prepare.py", line 603, in get_plan
    toil.start(start_job)
  File "/HOME/biosoft/cactus_env/lib/python3.6/site-packages/toil/common.py", line 823, in start
    return self._runMainLoop(rootJobDescription)
  File "/HOME/biosoft/cactus_env/lib/python3.6/site-packages/toil/common.py", line 1116, in _runMainLoop
    jobCache=self._jobCache).run()
  File "/HOME/biosoft/cactus_env/lib/python3.6/site-packages/toil/leader.py", line 228, in run
    self.innerLoop()
  File "/HOME/biosoft/cactus_env/lib/python3.6/site-packages/toil/leader.py", line 623, in innerLoop
    self.checkForDeadlocks()
  File "/HOME/biosoft/cactus_env/lib/python3.6/site-packages/toil/leader.py", line 655, in checkForDeadlocks
    totalRunningJobs = len(self.batchSystem.getRunningBatchJobIDs())
  File "/HOME/biosoft/cactus_env/lib/python3.6/site-packages/toil/batchSystems/abstractGridEngineBatchSystem.py", line 375, in getRunningBatchJobIDs
    batchIds = self.with_retries(self.worker.getRunningJobIDs)
  File "/HOME/biosoft/cactus_env/lib/python3.6/site-packages/toil/batchSystems/abstractGridEngineBatchSystem.py", line 438, in with_retries
    return operation(*args, **kwargs)
  File "/HOME/biosoft/cactus_env/lib/python3.6/site-packages/toil/batchSystems/torque.py", line 92, in getRunningJobIDs
    walltime = time.mktime(time.strptime(walltime, "%H:%M:%S"))
  File "/HOME/soft/cactpython/lib/python3.6/_strptime.py", line 559, in _strptime_time
    tt = _strptime(data_string, format)[0]
  File "/HOME/soft/cactpython/lib/python3.6/_strptime.py", line 362, in _strptime
    (data_string, format))
ValueError: time data '24:28:49' does not match format '%H:%M:%S'
```
As you can see from the output, the software cactus is actually running, however, there is an error. Do you have any suggestions for this problem? Any help will be much appreciated, Thanks.

Yawen

┆Issue is synchronized with this [Jira Task](https://ucsc-cgl.atlassian.net/browse/TOIL-953)
┆Issue Number: TOIL-953

**Comment by adamnovak** [comment]
Your problem is that you are running a job that runs for over 24 hours, on the Torque batch system. The bug is that we are using [`strptime()`](https://docs.python.org/3/library/datetime.html#datetime.datetime.strptime) to parse a duration, when it is intended to parse a time of day. It works for times of under 24 hours, but at 24 hours or more it starts to raise this error.

We need to change this line here:

https://github.com/DataBiosphere/toil/blob/70359d1fe4deed2edc60fb76a6d162bf319beec1/src/toil/batchSystems/torque.py#L91

We may need to write our own interval-parsing code, because I don't think there's anything like `strptime` for `timedelta`.

**Comment by lningyawen** [comment]
Thanks for your relpy! This bug has been bothering me for a long time. Using the cactus software to align genomes requires a lot of running resources and I don't think the calculations  on my data can be done in 24 hours. Therefore, I would like to ask what should I do to fix this problem?

**Comment by adamnovak** [comment]
I'm not sure if there's any way to make `cactus-prepare` chop your job into smaller pieces (@glennhickey?), but if you can't use a scheduler other than Torque, you would need to wait for us to fix the bug in Toil where we don't have a real duration parser (or fix it yourself), and then use the fixed Toil version (possibly installed from Git, if you don't want to wait for us to make a release).

We just got a new Toil team member today; I will see if he can take on fixing this.

**Comment by Hexotical** [comment]
I can probably take this on, might take a little bit for me to get set up, but i'll work on this today.

**Comment by lningyawen** [comment]
Hello

Sorry for getting back to you so late. Thank you both so much for your help! I have followed lucas' changes to `src/toil/lib/conversions.py` and `src/toil/batchSystems/torque.py` and have also modified my own python files in HPC, and I no longer get the error about `ValueError: time data '24:28:49' does not match format '%H:%M:%S'`. 
Everything seems to be working fine now. Thanks again for your help!

Yawen

**Comment by mr-c** [comment]
This has been fixed in #3708
